### PR TITLE
Hide nav bar when typing in POI chat

### PR DESCRIPTION
**Pull request by tberkane**

### CODE REVIEW

**Review by Bwananah — APPROVED** [review]
Good job ! Thanks !

**Comment by aman3014** [comment]
I'm not sure if I understand why this is a bug. If it is, why is it only being changed for the poi chat and not the friend chat?

**Comment by tberkane** [comment]
> I'm not sure if I understand why this is a bug. If it is, why is it only being changed for the poi chat and not the friend chat?

It's a bug because we don't want to have the nav bar go over the keyboard while typing in the chat. For example, the forum doesn't have this bug because when typing in the fields, the nav bar is hidden. The friend chat is not concerned by this because there is no nav bar in it anyways.

File: app/src/main/java/com/github/epfl/meili/messages/ChatLogActivity.kt [hunk 1]
```diff
@@ -47,13 +52,28 @@ class ChatLogActivity : PoiActivity(R.layout.activity_chat_log, R.id.chat_activi
 
         findViewById<RecyclerView>(R.id.recyclerview_chat_log).adapter = adapter
```
Comment (aman3014) [inline app/src/main/java/com/github/epfl/meili/messages/ChatLogActivity.kt:53]: We should store this as an attribute of the class.

File: app/src/main/java/com/github/epfl/meili/messages/ChatLogActivity.kt [hunk 2]
```diff
@@ -47,13 +52,28 @@ class ChatLogActivity : PoiActivity(R.layout.activity_chat_log, R.id.chat_activi
...
+        chatLogView.viewTreeObserver.addOnGlobalLayoutListener {
+            val r = Rect()
+            chatLogView.getWindowVisibleDisplayFrame(r)
+            val heightDiff: Int = chatLogView.rootView
+                .height - (r.bottom - r.top)
+            if (heightDiff > KEYBOARD_THRESHOLD) {
+                navigationBar.visibility = View.GONE
+            } else {
+                navigationBar.visibility = View.VISIBLE
+            }
```
Comment (aman3014) [inline app/src/main/java/com/github/epfl/meili/messages/ChatLogActivity.kt:66]: navigationBar.isVisible = heighDiff > KEYBOARD_THRESHOLD

**Review by aman3014 — COMMENTED** [review]
Okay, makes sense! Just a little change before we merge.

**Review by aman3014 — APPROVED** [review]

**Pull request merged**